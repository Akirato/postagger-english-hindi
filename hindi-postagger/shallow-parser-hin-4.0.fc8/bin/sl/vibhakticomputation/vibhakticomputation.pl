#!/usr/bin/perl

use Getopt::Long;
GetOptions("help!"=>\$help,"path=s"=>\$vibh_home,"input=s"=>\$input,"output=s",\$output);
print "Unprocessed by Getopt::Long\n" if $ARGV[0];
foreach (@ARGV) {
	print "$_\n";
	exit(0);
}

if($help eq 1)
{
	print "Vibhakti Computation  - Vibhakti Computation Version 2.3.2\n(14th July 2007 last modified on 9th August 2009)\n\n";
	print "usage : ./vibhakticomputation.pl --path=/home/vibhakticomputation-2.3.2 [-i inputfile|--input=\"input_file\"] [-o outputfile|--output=\"output_file\"] \n";
	print "\tIf the output file is not mentioned then the output will be printed to STDOUT\n";
	exit(0);
}

if($vibh_home eq "")
{
	print "Please Specify the Path as defined in --help\n";
	exit(0);

}

my $src=$vibh_home . "/src";
my $output_tmp=$vibh_home . "/OUTPUT.tmp";
require "$vibh_home/API/shakti_tree_api.pl";
require "$vibh_home/API/feature_filter.pl";
require "$src/ComputeTAM.pl";
require "$src/vibhakti_compute.pl";

if ($input eq "")
{
  $input="/dev/stdin";
}


&read_story($input);

$numBody = &get_bodycount();
for(my($bodyNum)=1;$bodyNum<=$numBody;$bodyNum++)
{

	$body = &get_body($bodyNum,$body);

# Count the number of Paragraphs in the story
	my($numPara) = &get_paracount($body);

#print STDERR "Paras : $numPara\n";

# Iterate through paragraphs in the story
	for(my($i)=1;$i<=$numPara;$i++)
	{

		my($para);
		# Read Paragraph
		$para = &get_para($i);


		# Count the number of sentences in this paragraph
		my($numSent) = &get_sentcount($para);
		# Iterate through sentences in the paragraph
		for(my($j)=1;$j<=$numSent;$j++)
		{

			# Read the sentence which is in SSF format
			my($sent) = &get_sent($para,$j);

			#Copy Vibhakti Info
			&ComputeVibhakti($sent,$vibh_home);

			#Compute TAM
			&ComputeTAM($sent,$vibh_home,$slang);
		}
	}
}

if($output eq "" )
{
	&printstory();
}

else
{
	&printstory_file("$output");
}
