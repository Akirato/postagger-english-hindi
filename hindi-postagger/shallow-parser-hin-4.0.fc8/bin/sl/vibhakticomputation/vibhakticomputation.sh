perl $SHALLOW_PARSER_HIN/bin/sl/vibhakticomputation/printinput.pl $1 > /tmp/vibhcomputeinput$$.tmp
#perl $setu/bin/sys/common/printinput.pl $1 > vibhakticomputationinput
perl $SHALLOW_PARSER_HIN/bin/sl/vibhakticomputation/vibhakticomputation.pl --path=$SHALLOW_PARSER_HIN/bin/sl/vibhakticomputation --input=/tmp/vibhcomputeinput$$.tmp
rm -fr /tmp/vibhcomputeinput$$.tmp
