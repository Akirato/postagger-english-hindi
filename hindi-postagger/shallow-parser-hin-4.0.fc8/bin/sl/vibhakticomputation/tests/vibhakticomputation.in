<Sentence id="1">
1	((	NP	<fs af='niyaMwraNa,,m,s,,0,0,' head="niyaMwraNa">
1.1	niyaMwraNa	NN	<fs af='niyaMwraNa,,m,s,,0,0,' name="niyaMwraNa">
1.2	,	SYM	
	))		
2	((	NP	<fs af='aMkuraNa,,m,s,,1,,' head="aMkuraNa">
2.1	aMkuraNa	NN	<fs af='aMkuraNa,,m,s,,1,,' name="aMkuraNa">
2.2	ke	PSP	<af='kA,,m,s,,1,,'>
2.3	kAraNa	PSP	<af='kAraNa,,m,s,,0,,'>
	))		
3	((	NP	<fs af='samaya,,m,s,,0,0,' head="samaya">
3.1	samaya	NN	<fs af='samaya,,m,s,,0,0,' name="samaya">
	))	
4	((	VGF	<fs af='ho,,m,p,any,1,,nA' head="hone">
4.1	hone	VNN	<fs af='ho,,m,p,any,1,,nA' name="hone">
	))
5	((	VGF	<fs af='ho,,m,p,any,1,nA,nA' head="hone">
5.1	hone	VNN	<fs af='ho,,m,p,any,1,nA,nA' name="hone">
5.2	vAlI	PSP	<fs af='vAlI,,f,s,,0,,'>
	))	
6	((	VGNN	<fs af='ho,,m,p,any,1,nA,nA' head="hone">
6.1	hone	VNN	<fs af='ho,,m,p,any,1,nA,nA' name="hone">
6.2	vAlI	PSP	<fs af='vAlI,,f,s,,0,,'>
	))	
7	((	VGNF	<fs af='ho,,m,p,any,1,nA,nA' head="hone">
7.1	hone	VNN	<fs af='ho,,m,p,any,1,nA,nA' name="hone">
7.2	vAlI	PSP	<fs af='vAlI,,f,s,,0,,'>
	))		
8	((	NP	<fs af='hAni,,f,s,,1,,' head="hAni">
8.1	hAni	NN	<fs af='hAni,,f,s,,1,,' name="hAni">
8.2	ko	PSP	<fs af='ko,,,,,,,'>
	))		
9	((	NP	<fs af='bIjopacAra,,,,,1,,' head="bIjopacAra">
9.1	uciwa	JJ	<fs af='uciwa,,any,any,,any,,'>
9.2	bIjopacAra	NN	<fs af='bIjopacAra,,,,,1,,' name="bIjopacAra">
9.3	xvArA	PSP	<fs af=xvArA,,,,,,,'>
	))		
10	((	VGF	<fs af='roka,,m,s,any,,yA,' head="roka">
10.1	rokA	VFM	<fs af='roka,,m,s,any,,yA,' name="roka">
10.2	jA	VAUX	<fs af='jA,,any,any,any,,0,'>
10.3	sakawa	VAUX	<fs af='saka,,f,s,any,,wA,'>
10.4	hE	VAUX	<fs af='hE,,any,s,m,,hE,'>
10.5	.	SYM	
	))	
11	((	VGF	<fs af='roka,,m,s,any,,yA,' head="roka">
11.1	rokA	VFM	<fs af='roka,,m,s,any,,yA,' name="roka">
12.2	.	SYM	
	))
</Sentence>	
