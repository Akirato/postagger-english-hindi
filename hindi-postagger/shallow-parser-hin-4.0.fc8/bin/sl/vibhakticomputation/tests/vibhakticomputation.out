<Sentence id="1">
1	((	NP	<fs af='niyaMwraNa,,m,s,,0,0,' head="niyaMwraNa">
1.1	niyaMwraNa	NN	<fs af='niyaMwraNa,,m,s,,0,0,' name="niyaMwraNa">
1.2	,	SYM	
	))		
2	((	NP	<fs af='aMkuraNa,,m,s,,1,0_kA_kAraNa,' head="aMkuraNa" vpos="vib1_2_3">
2.1	aMkuraNa	NN	<fs af='aMkuraNa,,m,s,,1,,' name="aMkuraNa">
	))		
3	((	NP	<fs af='samaya,,m,s,,0,0,' head="samaya">
3.1	samaya	NN	<fs af='samaya,,m,s,,0,0,' name="samaya">
	))		
4	((	VGF	<fs af='ho,,m,p,any,1,,nA' head="hone">
4.1	hone	VNN	<fs af='ho,,m,p,any,1,,nA' name="hone">
	))		
5	((	VGF	<fs af='ho,,m,s,any,1,nA_vAlI,nA' head="hone" vpos="tam1_2">
5.1	hone	VNN	<fs af='ho,,m,p,any,1,nA,nA' name="hone">
	))		
6	((	VGNN	<fs af='ho,,m,s,any,1,nA_vAlI,nA' head="hone" vpos="tam1_2">
6.1	hone	VNN	<fs af='ho,,m,p,any,1,nA,nA' name="hone">
	))		
7	((	VGNF	<fs af='ho,,m,s,any,1,nA_vAlI,nA' head="hone" vpos="tam1_2">
7.1	hone	VNN	<fs af='ho,,m,p,any,1,nA,nA' name="hone">
	))		
8	((	NP	<fs af='hAni,,f,s,,1,0_ko,' head="hAni" vpos="vib1_2">
8.1	hAni	NN	<fs af='hAni,,f,s,,1,,' name="hAni">
	))		
9	((	NP	<fs af='bIjopacAra,,,,,1,0_xvArA,' head="bIjopacAra" vpos="vib2_3">
9.1	uciwa	JJ	<fs af='uciwa,,any,any,,any,,'>
9.2	bIjopacAra	NN	<fs af='bIjopacAra,,,,,1,,' name="bIjopacAra">
	))		
10	((	VGF	<fs af='roka,,f,s,m,,yA_jA_saka+wA_hE,' head="roka" vpos="tam1_2_3_4">
10.1	rokA	VFM	<fs af='roka,,m,s,any,,yA,' name="roka">
10.2	.	SYM	
	))		
11	((	VGF	<fs af='roka,,m,s,any,,yA,' head="roka">
11.1	rokA	VFM	<fs af='roka,,m,s,any,,yA,' name="roka">
11.2	.	SYM	
	))		
</Sentence>
